gpusim.cpp

```diff
@@ -276,24 +287,37 @@ void GPUSimServer::searchAll(const Fingerprint& query, int results_requested,
 void GPUSimServer::incomingSearchRequest()
 {
     auto clientConnection = static_cast<QLocalSocket*>(sender());
-    QString dbname = clientConnection->property("dbname").toString();
 
     // Read incoming Fingerprint binary and put it in Fingerprint object
     QByteArray data = clientConnection->readAll();
     QDataStream qds(&data, QIODevice::ReadOnly);
 
+    int database_search_count;
+    qds >> database_search_count;
+    map<string, QString> dbname_to_key;
+    for(int i=0; i<database_search_count; i++) {
+        char* dbname;
+        char* dbkey;
+        qds >> dbname;
+        qds >> dbkey;
+        dbname_to_key[std::string(dbname)] = QString(dbkey);
```

Comment:
can you write a hash for QString so that this can be map<QString, QString>, since that's what you're using below anyways? Something like: https://stackoverflow.com/a/48527988

Reply:
Fair enough...now I need to figure out where/how to put that so it's accessible across C++ files...